compose/cli/main.py

```diff
@@ -1312,3 +1321,51 @@ def build_exec_command(options, container_id, command):
     args += [container_id]
     args += command
     return args
+
+
+def has_container_with_state(containers, state):
+    for container in containers:
+        states = {
+            'running': container.is_running,
+            'stopped': not container.is_running,
+            'paused': container.is_paused,
+        }
+        if state not in states:
+            raise UserError("Invalid state: %s" % state)
+        if states[state]:
+            return True
+    return False
+
+
+def filter_services(filters, services, project):
+    def should_include(service):
+        for f in filters:
+            if f == 'status':
+                containers = project.containers([service.name], stopped=True)
+                for status in filters[f]:
+                    if not has_container_with_state(containers, status):
+                        return False
+            elif f == 'key':
+                for key in filters[f]:
+                    if key == 'image' or key == 'build':
+                        if key not in service.options:
+                            return False
+                    else:
+                        raise UserError("Invalid option: %s" % key)
+            else:
+                raise UserError("Invalid filter: %s" % f)
+        return True
+
+    return filter(should_include, services)
+
+
+def build_filters(args):
+    filters = {}
+    for arg in args:
+        if '=' not in arg:
+            raise UserError("Arguments to --filter should be in form KEY=VAL")
+        key, val = arg.split('=', 1)
+        if key not in filters:
+            filters[key] = []
+        filters[key].append(val)
```

Comment:
Is there a use-case for having multiple values for the same filter? In other words, when would a user possibly need to do something like `docker-compose ps --filter state=paused --filter state=running`?

If there's nothing compelling, we could simplify the code a lot by allowing only one value per filter.

Reply:
`docker ps supports` multiple filters of the same kind, so it might be worth implementing it here for symmetry reasons, too. But I'm OK with a simpler solution. In that case, there should be an error message for duplicate filter usage.

Reply:
Simplified the code, now you can only have one `--filter` argument.